src/proto/streams/streams.rs

```diff
@@ -322,6 +322,10 @@ where
         let last_stream_id = frame.last_stream_id();
         let err = frame.reason().into();
 
+        if actions.recv.max_stream_id() < last_stream_id {
```

Comment:
I got a bit confused with this, until I went digging into `recv` to read the comments about `max_stream_id`. Whatcha think if there was a comment right here just saying to the effect of "if a new GOAWAY has a higher stream id than a previous GOAWAY, that's bad"?

Reply:
:+1: